lib/matplotlib/tests/test_cm.py

```diff
@@ -0,0 +1,27 @@
+import matplotlib.cm as mcm
+import warnings
+import pytest
+
+def test_double_register_cmap():
+    with pytest.raises(ValueError):
+        mcm.register_cmap('viridis', mcm.get_cmap('viridis'))
+
+
+def test_register_cmap():
+    mcm.register_cmap('viridis'[::-1], mcm.get_cmap('viridis'))
+    assert mcm.get_cmap('viridis'[::-1]) is mcm.get_cmap('viridis')
```

Comment:
I find this very hard to read; in the sense of you cannot directly see what's going on since there is only lots of viridis in the game. If naming it e.g. "newname" everything would be a lot clearer.